lib/msf/core/post/windows/services.rb

```diff
@@ -181,32 +183,61 @@ def each_service(&block)
   # @todo Rewrite to allow operating on a remote host
   #
   def service_list
-    if load_extapi
-      return session.extapi.service.enumerate
-    else
-      serviceskey = "HKLM\\SYSTEM\\CurrentControlSet\\Services"
-      a =[]
-      services = []
-      keys = registry_enumkeys(serviceskey)
-      keys.each do |s|
-        if a.length >= 10
-          a.first.join
-          a.delete_if {|x| not x.alive?}
+    SERVICES.clear
```

Comment:
Please don't modify constants. The way you're using this doesn't seem like you need anything more than just a local variable.